internal/app/management/run.go

```diff
@@ -34,18 +38,47 @@ func Run(config *types.Configuration) {
 		panic(fmt.Errorf("failed to initialize the trace management server: %+v", err))
 	}
 
-	lis, err := net.Listen("tcp", fmt.Sprintf(":%d", config.Port))
+	var options []grpc.ServerOption
+
+	if config.CertFile != "" && config.KeyFile != "" && config.CACertFile != "" {
```

Comment:
Nit: could use the same `func (t *TLSCerts) Available()` function here as you do in the CLI as it's cleaner.